src/contracts/OriginARM.sol

```diff
@@ -0,0 +1,101 @@
+    event ClaimOriginWithdrawals(uint256[] requestIds, uint256 amountClaimed);
+
+    /// @param _otoken The address of the Origin token. eg OETH or OS
+    /// @param _liquidityAsset The address of the liquidity asset. eg WETH or wS
+    /// @param _vault The address of the Origin Vault
+    /// @param _claimDelay The delay in seconds before a user can claim a redeem from the request
+    constructor(
+        address _otoken,
+        address _liquidityAsset,
+        address _vault,
+        uint256 _claimDelay,
+        uint256 _minSharesToRedeem
+    ) AbstractARM(_liquidityAsset, _otoken, _liquidityAsset, _claimDelay, _minSharesToRedeem) {
+        vault = _vault;
+
+        _disableInitializers();
+    }
+
+    /// @notice Initialize the storage variables stored in the proxy contract.
+    /// The deployer that calls initialize has to approve the ARM's proxy contract to transfer 1e12 liquid assets.
+    /// @param _name The name of the liquidity provider (LP) token.
+    /// @param _symbol The symbol of the liquidity provider (LP) token.
+    /// @param _operator The address of the account that can request and claim Lido withdrawals.
+    /// @param _fee The performance fee that is collected by the feeCollector measured in basis points (1/100th of a percent).
+    /// 10,000 = 100% performance fee
+    /// 1,500 = 15% performance fee
+    /// @param _feeCollector The account that can collect the performance fee
+    /// @param _capManager The address of the CapManager contract
+    function initialize(
+        string calldata _name,
+        string calldata _symbol,
+        address _operator,
+        uint256 _fee,
+        address _feeCollector,
+        address _capManager
+    ) external initializer {
+        _initARM(_operator, _name, _symbol, _fee, _feeCollector, _capManager);
+    }
+
+    /**
+     * @notice Request a withdrawal of oTokens from the Origin Vault.
+     * @param amount The amount of oTokens to withdraw from the Origin Vault.
+     * @return requestId The ID of the Origin Vault withdrawal request.
+     */
+    function requestOriginWithdrawal(uint256 amount) external onlyOperatorOrOwner returns (uint256 requestId) {
+        (requestId,) = IOriginVault(vault).requestWithdrawal(amount);
+
+        // Increase the outstanding withdrawal amount from the Origin Vault
+        vaultWithdrawalAmount += amount;
+
+        emit RequestOriginWithdrawal(amount, requestId);
+    }
+
+    /**
+     * @notice Claim multiple previously requested withdrawals from the Origin Vault.
+     * The caller should check the withdrawal has passed the withdrawal delay
+     * and there is enough liquidity in the Vualt.
+     * @param requestIds The request IDs of the withdrawal requests.
+     * @param amountClaimed The total amount claimed across all withdrawal requests.
+     */
+    function claimOriginWithdrawals(uint256[] calldata requestIds) external returns (uint256 amountClaimed) {
```

Comment:
Why do we run out of liquidity? Mostly when we are the best price on the market. Which means that there might be room to charge more money. 

This suggests that in an ideal world, whenever we re-added liquidity, from whatever source, be it moving funds from a lending platform, Lido withdraws, or claiming from a vault, that we would want this to be done after potential price change. In order to enforce this, we would need to have these funds moving in functions only be called by the owner/operator.

Reply:
I don't think there's any harm in running it the current way. We certainly can choose to do so. it's just there's a possibility of trading getting off earning more money in exchange for more work on the backend side.